File: sdk/php/demonck/fxtaskbar/FXTaskBarNativeWindows.php
<?php

namespace php\demonck\fxtaskbar;

use php\gui\UXScene;

/**
 * FXTaskBarNativeWindows класс для управления прогрессом на панели задач.
 */
class FXTaskBarNativeWindows
{
    /**
     * Конструктор класса.
     *
     * @param Scene $scene Объект сцены, используемый для получения окна.
     */
    public function __construct(UXScene $scene)
    {
    }

    /**
     * Проверяет поддержку.
     *
     * @return bool Возвращает true, если поддержка есть.
     */
    public static function isSupport()
    {
    }

    /**
     * Проверяет отсутствие поддержку.
     *
     * @return bool Возвращает true, если поддержка отсутствует.
     */
    public static function isNotSupport()
    {
    }

    /**
     * Показывает неопределенный прогресс на панели задач.
     *
     * @return void
     */
    public function showIndeterminateProgress()
    {
    }

    /**
     * Показывает пользовательский прогресс на панели задач.
     *
     * @param int $l1 Текущее значение прогресса.
     * @param int $l2 Общее значение прогресса.
     * @param string $type Тип прогресса (NORMAL, ERROR и т. д.).
     * @return void
     */
    public function showCustomProgress(int $l1, int $l2, string $type)
    {
    }

    /**
     * Останавливает отображение прогресса на панели задач.
     *
     * @return void
     */
    public function stopProgress()
    {
    }

    /**
     * Показывает прогресс с полной ошибкой на панели задач.
     *
     * @return void
     */
    public function showFullErrorProgress()
    {
    }

    /**
     * Показывает прогресс в состоянии паузы на панели задач.
     *
     * @return void
     */
    public function showFullPausedProgress()
    {
    }

    /**
     * Показывает нормальный полный прогресс на панели задач.
     *
     * @return void
     */
    public function showFullNormalProgress()
    {
    }
}

File: src-jvm/main/java/org/develnext/jphp/ext/fxtaskbar/nativewindows/evodevelopers/classes/FXTaskBarNativeWindows.java
package org.develnext.jphp.ext.fxtaskbar.nativewindows.evodevelopers.classes;

import com.nativejavafx.taskbar.TaskbarProgressbar;
import javafx.application.Application;
import javafx.application.Platform;
import javafx.scene.Scene;
import javafx.stage.Stage;
import org.develnext.jphp.ext.fxtaskbar.nativewindows.evodevelopers.FXTaskBarExtension;
import php.runtime.annotation.Reflection;
import php.runtime.env.Environment;
import php.runtime.lang.BaseObject;
import php.runtime.reflection.ClassEntity;

@Reflection.Name("FXTaskBarNativeWindows")
@Reflection.Namespace(FXTaskBarExtension.NS)
public class FXTaskBarNativeWindows extends BaseObject {

    private Stage _stage;

    protected FXTaskBarNativeWindows(ClassEntity entity) {
        super(entity);
    }

    public FXTaskBarNativeWindows(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Reflection.Signature
    public void __construct(Scene scene) {
        _stage = (Stage) scene.getWindow();
    }

    @Reflection.Signature
    public static boolean isSupport()
    {
        return TaskbarProgressbar.isSupported();
    }

    @Reflection.Signature
    public static boolean isNotSupport()
    {
        return TaskbarProgressbar.isNotSupported();
    }

    @Reflection.Signature
    public void showIndeterminateProgress()
    {
        TaskbarProgressbar.showIndeterminateProgress(_stage);
    }

    @Reflection.Signature
    public void showCustomProgress(long l1, long l2, String type)
    {
        TaskbarProgressbar.showCustomProgress(_stage, l1, l2, getTypeForString(type));
    }

    @Reflection.Signature
    public void stopProgress()
    {
        TaskbarProgressbar.stopProgress(_stage);
    }

    @Reflection.Signature
    public void showFullErrorProgress()
    {
        TaskbarProgressbar.showFullErrorProgress(_stage);
    }

    @Reflection.Signature
    public void showFullPausedProgress()
    {
        TaskbarProgressbar.showFullPausedProgress(_stage);
    }

    @Reflection.Signature
    public void showFullNormalProgress()
    {
        TaskbarProgressbar.showFullNormalProgress(_stage);
    }

    private TaskbarProgressbar.Type getTypeForString(String s)
    {
        TaskbarProgressbar.Type type = TaskbarProgressbar.Type.NORMAL;

        switch (s){
            case ("TaskbarProgressbar.Type.ERROR"):
                type = TaskbarProgressbar.Type.ERROR;
                break;
            case ("TaskbarProgressbar.Type.INDETERMINATE"):
                type = TaskbarProgressbar.Type.INDETERMINATE;
                break;
            case ("TaskbarProgressbar.Type.NORMAL"):
                type = TaskbarProgressbar.Type.NORMAL;
                break;
            case ("TaskbarProgressbar.Type.PAUSED"):
                type = TaskbarProgressbar.Type.PAUSED;
                break;
            default:
                 type = TaskbarProgressbar.Type.NORMAL;
        }

        return type;
    }

}
